NineChronicles.Headless/GraphTypes/TransactionHeadlessQuery.cs

```diff
@@ -185,6 +181,133 @@ public TransactionHeadlessQuery(StandaloneContext standaloneContext)
+                    }
+
+                    string plainValueString = context.GetArgument<string>("plainValue");
+                    var plainValue = new Bencodex.Codec().Decode(ByteUtil.ParseHex(plainValueString));
+#pragma warning disable 612
+                    var action = new NCAction();
+#pragma warning restore 612
+                    action.LoadPlainValue(plainValue);
+
+                    var publicKey = new PublicKey(ByteUtil.ParseHex(context.GetArgument<string>("publicKey")));
+                    Address signer = publicKey.ToAddress();
+                    long nonce = context.GetArgument<long?>("nonce") ?? blockChain.GetNextTxNonce(signer);
+                    Transaction<NCAction> unsignedTransaction =
+                        Transaction<NCAction>.CreateUnsigned(nonce, publicKey, blockChain.Genesis.Hash, new[] { action });
+                    return unsignedTransaction.Serialize(false);
+                });
+
+            Field<NonNullGraphType<ByteStringType>>(
+                name: "signTransaction",
+                arguments: new QueryArguments(
+                    new QueryArgument<NonNullGraphType<StringGraphType>>
+                    {
+                        Name = "unsignedTransaction",
+                        Description = "The hexadecimal string of unsigned transaction to attach the given signature."
+                    },
+                    new QueryArgument<NonNullGraphType<StringGraphType>>
+                    {
+                        Name = "signature",
+                        Description = "The hexadecimal string of signature of the given unsigned transaction."
+                    }
+                ),
+                resolve: context =>
+                {
+                    byte[] signature = ByteUtil.ParseHex(context.GetArgument<string>("signature"));
+                    Transaction<NCAction> unsignedTransaction =
+                        Transaction<NCAction>.Deserialize(
+                            ByteUtil.ParseHex(context.GetArgument<string>("unsignedTransaction")),
+                            false);
+                    Transaction<NCAction> signedTransaction = new Transaction<NCAction>(
+                        unsignedTransaction.Nonce,
+                        unsignedTransaction.Signer,
+                        unsignedTransaction.PublicKey,
+                        unsignedTransaction.GenesisHash,
+                        unsignedTransaction.UpdatedAddresses,
+                        unsignedTransaction.Timestamp,
+                        unsignedTransaction.Actions,
+                        signature);
+                    signedTransaction.Validate();
+
+                    return signedTransaction.Serialize(true);
+                }
+            );
+
+            Field<NonNullGraphType<TxIdType>>(
+                name: "stageTx",
+                description: "Add a new transaction to staging and return TxId",
```

Comment:
IMO, I suggest moving this query to mutation because I think it modifies server-side data (`BlockChain<T>`).

https://graphql.org/learn/queries/#mutations

> Most discussions of GraphQL focus on data fetching, but any complete data platform needs a way to modify server-side data as well.
>
> In REST, any request might end up causing some side-effects on the server, but by convention it's suggested that one doesn't use GET requests to modify data. GraphQL is similar - technically any query could be implemented to cause a data write. However, it's useful to establish a convention that any operations that cause writes should be sent explicitly via a mutation.